src/cmd/project_init.js

```diff
@@ -65,49 +74,87 @@ export class ProjectInitCommandSite extends CommandSite {
 
 	}
 
+	error(error) {
+		throw error;
+}
 }
 
 /**
  * Implements the project initialization command.
  */
 export class ProjectInitCommand extends Command {
 
+	static templateFile(name) {
+		return path.join(__dirname, 'templates', 'project', name);
+	}
+
+	static expandTemplate(fs, templateName, data) {
+		const readFile = promisify(fs.readFile);
+		return readFile(ProjectInitCommand.templateFile(templateName), 'utf-8')
+			.then(content => {
+				return require('underscore').template(content)(data);
+			});
+	}
+
+	/**
+	 *
+	 * @param name
+	 * @returns {object} - key valid indicates if validation passed
+	 *                     key errors is an object with pairs of invalid field names and error messages
+	 */
+	validateName(name) {
+		return validateField('name', name);
+	}
+
 	/**
 	 *
 	 * @param {object} state The current conversation state.
 	 * @param {ProjectInitCommandSite} site external services.
 	 * @returns {Promise} To run the project initialization command.
 	 */
 	run(state, site) {
-		let directory, filesystem;
+		let directory, filesystem, name;
 
 		return Promise.resolve()
-			.then(() => Promise.resolve(site.directory()))
+		.then(() => Promise.resolve(site.name()))
```

Comment:
yeah, totally agree it's not needed. I think it was just a habit to indicate that these site callbacks can return a value or a promise.